frontend/src/scenes/insights/InsightTabs/RetentionTabHorizontal.tsx

```diff
@@ -0,0 +1,188 @@
+                        <Col>
+                            <Button ref={node} data-attr="retention-action" onClick={() => setOpen(!open)}>
+                                <PropertyKeyInfo value={selectedCohortizingEvent} disablePopover />
+                                <DownOutlined
+                                    className="svg-fix"
+                                    style={{ marginRight: '-6px', marginTop: 2, color: '#bdbdbd', fontSize: '1.3em' }}
+                                />
+                            </Button>
+                            <ActionFilterDropdown
+                                open={open}
+                                logic={entityLogic as any}
+                                openButtonRef={node}
+                                onClose={() => setOpen(false)}
+                            />
+                        </Col>
+                        <Col>
+                            <div style={{ display: '-webkit-inline-box', flexWrap: 'wrap' }}>
+                                <Select
+                                    value={retentionOptions[filters.retention_type]}
+                                    onChange={(value): void => setFilters({ retention_type: value as RetentionType })}
+                                    dropdownMatchSelectWidth={false}
+                                >
+                                    {Object.entries(retentionOptions).map(([key, value]) => (
+                                        <Select.Option key={key} value={key}>
+                                            {value}
+                                            <Tooltip placement="right" title={retentionOptionDescriptions[key]}>
+                                                <InfoCircleOutlined className="info-indicator" />
+                                            </Tooltip>
+                                        </Select.Option>
+                                    ))}
+                                </Select>
+                            </div>
+                        </Col>
+                        <Col>grouped by</Col>
+                        <Col>
+                            <Select
+                                value={filters.period}
+                                onChange={(value): void => setFilters({ period: value })}
+                                dropdownMatchSelectWidth={false}
+                            >
+                                {dateOptions.map((period) => (
+                                    <Select.Option key={period} value={period}>
+                                        {period}
+                                    </Select.Option>
+                                ))}
+                            </Select>
+                        </Col>
+                        <Col>...</Col>
+                    </Row>
+                    <Row gutter={8} align="middle" className="mt">
+                        <Col>... and then came back and did</Col>
```

Comment:
Could it be `who came back and did?` Feel like repeating the word `who` is helpful to understanding what's going on.

Reply:
Actually had it that way first, but thought the "and" read nicer as a single sentence, but agreed here. Updating.